### DISCUSSION TITLE
Secured Submenu back jumps to root menu

### DISCUSSION THREAD

**Discussion by bambo1543** [davetcc/tcMenu#449] [General]
Hello,
when I click back on a SubSubmenu it dosen't navigate to the parent Submenu but to the root menu.

I'm using tcmenu 4.2.1 and set a AuthenticationManager:

`class MyAuthenticationManager : public AuthenticationManager {
public:
    explicit MyAuthenticationManager(AuthenticationManagerType authType) : AuthenticationManager(authType) {}

    bool isAuthenticated(const char *connectionName, const char *authResponse) override {
        serdebug3("isAuthenticated", connectionName, authResponse)
        return true;
    }

    bool addAdditionalUUIDKey(const char *connectionName, const char *uuid) override {return false; }

    bool doesPinMatch(const char *doesPinMatch) override {
        serdebug2("PIN: ", preferences.getString(PIN, "n/a"))
        return preferences.getString(PIN, "1234").equals(doesPinMatch);
    }
};
MyAuthenticationManager myAuthenticationManager(AUTHENTICATION_IN_FLASH);
setup() {
...
 menuMgr.setAuthenticator(&myAuthenticationManager);
...
}
`
My menu is:
Root -> Settings (secured) -> Wifi
When I click in Wifi-Menu back, it navigates to Root instead of Settings.

What I'm doing wrong. Thanks for help.
Andreas

**Comment by davetcc** [comment]
I’d need to confirm myself but I think that is expected behaviour to completely reset navigation on entering a secure menu.
